### PR TITLE
Add bitshift optimizer tests

### PR DESCRIPTION
**Pull request by Marenz**
Those tests required issue #4262 which was finished recently.

### CODE REVIEW

**Comment by Marenz** [comment]
Sorry, this contains more than it should, will fix

**Comment by Marenz** [comment]
Okay, this seems to fail because #4262 was only done for syntaxTests not for optimizer tests.

**Comment by chriseth** [comment]
Can you rebase on top of #6429 and re-try?

**Comment by Marenz** [comment]
rebased and adjusted

File: test/libyul/yulOptimizerTests/expressionSimplifier/remove_redundant_shift_masking.yul [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{
+    let a := and(0xff, shr(248, calldataload(0)))
+    let b := and(shr(248, calldataload(0)), 0xff)
+}
```
Comment (ekpyron) [inline test/libyul/yulOptimizerTests/expressionSimplifier/remove_redundant_shift_masking.yul:6]: Maybe add the edge cases, in which the mask isn't redundant?

**Comment by Marenz** [comment]
Added non-redundant edge cases

**Comment by chriseth** [comment]
Please rebase.

**Comment by Marenz** [comment]
Rebased

**Comment by chriseth** [comment]
Still not properly rebased.

**Comment by chriseth** [comment]
Ah sorry, I thought the other one is already merged!

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
Works locally, so I'll merge and break develop ;)

**Pull request merged**